modules/apps/27-interchain-accounts/host/keeper/msg_server.go

```diff
@@ -0,0 +1,37 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/host/types"
+	ibcerrors "github.com/cosmos/ibc-go/v7/modules/core/errors"
+)
+
+var _ types.MsgServer = (*msgServer)(nil)
+
+type msgServer struct {
+	*Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the ICS27 host MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+// UpdateParams updates the host submodule's params.
+func (m msgServer) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	if m.authority != msg.Authority {
+		return nil, errors.Wrapf(ibcerrors.ErrInvalidAddress, "invalid authority; expected %s, got %s", m.authority, msg.Authority)
```

Comment:
I wonder if we should have a `ibcerrors.ErrInvalidAuthority`?

Reply:
done